ra2ce/analysis/damages/damage_functions/damage_fraction_uniform.py

```diff
@@ -19,30 +19,34 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
 
+from __future__ import annotations
 
 import logging
+from dataclasses import dataclass
 from pathlib import Path
 
 import pandas as pd
+from scipy.interpolate import interp1d
 
 
+@dataclass(kw_only=True)
 class DamageFractionUniform:
     """
     Uniform: assuming the same curve for
     each road type and lane numbers and any other metadata
 
-
     self.raw_data (pd.DataFrame) : Raw data from the csv file
     self.data (pd.DataFrame) : index = hazard severity (e.g. flood depth); column 0 = damage fraction
-
     """
 
-    def __init__(self, name=None, hazard_unit=None):
-        self.name = name
-        self.hazard_unit = hazard_unit
-        self.interpolator = None
+    name: str = None
+    hazard_unit: str = None
+    data: pd.DataFrame = None
+    origin_path: Path = None
+    interpolator: interp1d = None
 
-    def from_csv(self, path: Path, sep=",") -> None:
+    @classmethod
+    def from_csv(cls, path: Path, sep=",") -> DamageFractionUniform:
```

Comment:
Missing typehint, also remember we agreed on removing default values for arguments, this way you force explicit declaration on the callers level.
```suggestion
    def from_csv(cls, path: Path, sep: str =",") -> DamageFractionUniform:
```

Reply:
Also, I'm not sure if I wrote the `path` argument, but it might be better to have some other name to avoid confussion with the library name, so something as `csv_path` instead of just `path`.

Reply:
Adjusted